networking/src/p2p/peer.rs

```diff
@@ -206,6 +208,192 @@ struct Network {
     socket_address: SocketAddr,
 }
 
+const THROTTLING_QUOTA_NUM: usize = 18;
+
+const THROTTLING_QUOTA_STRS: [&str; THROTTLING_QUOTA_NUM] = [
+    "Disconnect",
+    "Advertise",
+    "SwapRequest",
+    "SwapAck",
+    "Bootstrap",
+    "GetCurrentBranch",
+    "CurrentBranch",
+    "Deactivate",
+    "GetCurrentHead",
+    "CurrentHead",
+    "GetBlockHeaders",
+    "BlockHeader",
+    "GetOperations",
+    "Operation",
+    "GetProtocols",
+    "Protocol",
+    "GetOperationsForBlocks",
+    "OperationsForBlocks",
+];
+
+const THROTTLING_QUOTA_RESET_MS_DEFAULT: u64 = 5000; // 5 secs
+
+lazy_static::lazy_static! {
+    static ref THROTTLING_QUOTA_DISABLE: bool = {
+        match std::env::var("THROTTLING_QUOTA_DISABLE") {
+            Ok(v) => v.parse::<bool>().unwrap_or(false),
+            _ => false,
+        }
+    };
+
+    /// Quota reset period, in ms
+    static ref THROTTLING_QUOTA_RESET_MS: u64 = {
+        match std::env::var("THROTTLING_QUOTA_RESET_MS") {
+            Ok(v) => v.parse().unwrap_or(THROTTLING_QUOTA_RESET_MS_DEFAULT),
+            _ => THROTTLING_QUOTA_RESET_MS_DEFAULT,
+        }
+    };
+
+    /// Quota for tx/rx messages per [THROTTLING_QUOTA_RESET_MS]
+    static ref THROTTLING_QUOTA_MAX: [(isize, isize); THROTTLING_QUOTA_NUM] = {
+        let mut default = [
+            (1, 1), // Disconnect
+            (1, 1), // Advertise
+            (10, 10), // SwapRequest
+            (10, 10), // SwapAck
+            (1, 1), // Bootstrap
+            (10, 500), // GetCurrentBranch
```

Comment:
Here we accept 500 `GetCurrentBranch` messages. But below for `CurrentBranch` we have a quota of `200`. Maybe they should be equal? Here we will accept 500 get requests, but only respond to 200. Same for other kinds of messages.

Reply:
According to my measurements, there is an asymmetry in messages rate in Tezos network (https://viable-systems.slack.com/archives/C01SGFZ5L5P/p1626424648216800).

Reply:
@binier Anyway, that symmetry does make sense, so updated that.